source/service.go

```diff
@@ -121,20 +122,38 @@ func (sc *serviceSource) endpointsFromTemplate(svc *v1.Service) ([]*endpoint.End
 		return nil, fmt.Errorf("failed to apply template on service %s: %v", svc.String(), err)
 	}
 
+	ttl := getTTLFromAnnotations(svc.Annotations)
 	hostname := buf.String()
 	for _, lb := range svc.Status.LoadBalancer.Ingress {
 		if lb.IP != "" {
 			//TODO(ideahitme): consider retrieving record type from resource annotation instead of empty
-			endpoints = append(endpoints, endpoint.NewEndpoint(hostname, lb.IP, ""))
+			endpoints = append(endpoints, endpoint.NewEndpointWithTTL(hostname, lb.IP, "", ttl))
 		}
 		if lb.Hostname != "" {
-			endpoints = append(endpoints, endpoint.NewEndpoint(hostname, lb.Hostname, ""))
+			endpoints = append(endpoints, endpoint.NewEndpointWithTTL(hostname, lb.Hostname, "", ttl))
 		}
 	}
 
 	return endpoints, nil
 }
 
+func getTTLFromAnnotations(annotations map[string]string) endpoint.TTL {
+	ttlAnnotation, exists := annotations[ttlAnnotationKey]
+	if !exists {
+		return endpoint.TTL{IsConfigured: false}
+	}
+	ttlValue, err := strconv.ParseInt(ttlAnnotation, 10, 64)
+	if err != nil {
+		log.Warnf("%v is not a valid TTL value", ttlAnnotation)
+		return endpoint.TTL{IsConfigured: false}
+	}
+	if ttlValue < 0 {
+		log.Warnf("TTL must be a non-negative integer", ttlAnnotation)
+		return endpoint.TTL{IsConfigured: false}
```

Comment:
better return an error that you define:

errIllegalValue := errors.New("invalid TTL, must be within [10, 65535]") // please look up min and max ttl from rfc